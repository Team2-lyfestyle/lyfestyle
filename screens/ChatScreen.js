import React from 'react';
import {
  Image,
  Platform,
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
  SafeAreaView,
  Animated,
  Easing,
  RefreshControl,
  ActivityIndicator,
  LayoutAnimation,
  UIManager,
} from 'react-native';
import { GiftedChat } from 'react-native-gifted-chat';

function getChatUpdate() {
  return [
    {
      _id: 1,
      text: 'Hello',
      createdAt: new Date(),
      user: { _id: 2 }
    },
    {
      _id: 2,
      text: 'World',
      createdAt: new Date(),
      user: { _id: 2 }
    },
  ]
}

/*
navigation parameters:
  'chatSessionId'
*/
export default class ChatScreen extends React.Component {
  constructor(props) {
    super(props);
    this.props = props;

    // remember to bind class methods
    this.getMessages = this.getMessages.bind(this); 
    this.deleteLocalMessages = this.deleteLocalMessages.bind(this);
    
    this.messageCount = 0;    // Variable used to assign _id to messages

    this.state = {
      messages: [],
      dataIsLoaded: false,
    }

    this.otherUser = {
      _id: 2,
      name: "",
      avatar: 'https://placeimg.com/140/140/any',
      username: "",
    }
  }

  async componentDidMount() {
    
    /*
    this.thisUser = {
      _id: 1,
      name: tok.user.name,
      avatar: 'https://placeimg.com/140/140/any', // placeholder for actual avatar
      username: tok.user.username,
      token: tok.token,
    */
    
  }

  componentWillUnmount() {

  }

  onSend(messages = []) {
    // First, append new message to GiftedChat
    console.log(messages);
    this.setState(previousState => ({
      messages: GiftedChat.append(previousState.messages, messages),
    }));
  }

  getMessages() {
    request = {
      username: this.thisUser.username,
      token: this.thisUser.token,
      from: this.otherUser.username,
    }
    
  }

  deleteLocalMessages() {
    this.setState({
      messages: [],
    });
    this.messageCount = 0;
  }

  render() {
    if (!this.state.componentHasLoaded) {
      return (
        <SafeAreaView style={{flex: 1}}>
          <Text>Loading</Text>
          <ActivityIndicator />
        </SafeAreaView>
      )
    }
    else {
      return (
        <View style={ { flex: 1 } }>
          <GiftedChat 
            messages={this.state.messages}
            onSend={messages => this.onSend(messages)} 
            user={this.thisUser}
          />
        </View>
      );
    }
  }
}