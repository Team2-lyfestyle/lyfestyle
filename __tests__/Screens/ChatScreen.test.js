import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ChatScreen from '../../screens/ChatScreen';

import renderer from 'react-test-renderer';

describe('<ChatScreen />', () => {
  it('Should render the <ChatScreen /> component', () => {
    const wrapper = shallow(<ChatScreen />);
    expect(wrapper.find(ChatScreen)).toMatchSnapshot();
  });
});


