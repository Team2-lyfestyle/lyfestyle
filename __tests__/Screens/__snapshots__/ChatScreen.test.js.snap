// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatScreen /> Should render the <ChatScreen /> component 1`] = `null`;
